txtorcon/endpoints.py

```diff
@@ -667,6 +667,7 @@ class TorClientEndpoint(object):
     socks_ports_to_try = [9050, 9150]
 
     def __init__(self, host, port,
+                 socks_endpoint=None,
                  socks_hostname=None, socks_port=None,
                  socks_username=None, socks_password=None,
```

Comment:
Should be able to get rid of socks_hostname + socks_port too then, right? (i.e. it's `socks_endpoint=TCP4ClientEndpoint(host, port)` instead of socks_hostname=host and socks_port=port right?)